pkg/generate/app/cmd/newapp.go

```diff
@@ -919,7 +973,21 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 		return nil, err
 	}
 
-	if err := c.resolve(components); err != nil {
+	var imageComp app.ComponentReference
+	imageComp, repositories, err = c.addImageSource(repositories)
```

Comment:
@smarterclayton near as i can tell (i got this bit from @csrwng so he may have a better reason), it's done this way because the compRef returned from resolving an imagesource argument needs to be handled specially.  We still need to "resolve" it, but we do not want to send it through the rest of the new-app logic which would, for example, create a deploymentconfig for it.

So we need to pass that compRef to resolve(), but not to any of the other logic.  

Making that happen via the validate() method and signature would be messy.  At a minimum i think we'd have to call resolve separately from within validate().

Should this whole thing be refactored?  probably. Is this the reason to do it or the PR to do it in?  i don't think so.